src/queries/lendr-delegation-contract.ts

```
@@ -0,0 +1,507 @@
+  });
+
+  // Read contract functions - Core contract data
+  const { data: owner, refetch: refetchOwner } = useReadContract({
+    address: LENDR_DELEGATION_CONTRACT_ADDRESS,
+    abi: LENDR_DELEGATION_ABI,
+    functionName: 'i_owner',
+  });
+
+  // Helper functions for common read operations
+  const getDelegation = (nftContract: `0x${string}`, tokenId: bigint) => {
+    const { data: delegation, refetch } = useReadContract({
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)